nsot/api/views.py

```diff
@@ -504,23 +519,41 @@ def next_network(self, request, pk=None, site_pk=None, *args, **kwargs):
         prefix_length = params.get('prefix_length')
         num = params.get('num')
         strict = qpbool(params.get('strict_allocation', False))
-        networks = network.get_next_network(
-            prefix_length, num, strict, as_objects=False
-        )
-
-        return self.success(networks)
+        if request.method == 'POST':
+            state = 'reserved' if qpbool(params.get('reserve', False)) else 'allocated'
```

Comment:
These state strings are set as constants on the Network model, which you should use just in case we ever change the string in the future:
https://github.com/dropbox/nsot/blob/067699cdb28b3122cb36df613a79e208fb37fb9a/nsot/models.py#L690-L693